src/app/pages/FrontPage/index.stories.jsx

```diff
@@ -40,9 +30,16 @@ const DataWrapper = ({ service, variant, children }) => {
   const [data, setData] = useState();
 
   useEffect(() => {
-    preprocess(serviceDatasets[service][variant], preprocessorRules).then(
-      setData,
-    );
+    const fetchData = async () => {
+      fetchMock.getOnce(
```

Comment:
Exporting the transformer (`processJson`) as an additional named export may make our lives easier for situations like this

iirc there are also cypress files that want to use the transformers

Reply:
I don't really want to expose the functions that make up `getInitialData` because I want this function to be easy to refactor in future PRs. I'm trying to avoid the preprocessor pattern or `processJson` to be a thing or part of the `getInitialData` API.

I [made an example](https://github.com/bbc/simorgh/compare/replace-preprocessor-with-pipe-for-page-data...radio-page-almost-finished-example?expand=1#diff-86378fd076f9b9d13ecefad26f25c47f) of where I'm trying to get to with the refactor. I used the radio page as an example because it's the simplest route. You can see that we don't need to process the whole page data returned from Ares. Just pick the bits we need and alter the data as needed then pass down strictly only what the page component needs to render. This takes complexity out of the components and makes defining and parsing prop types much easier.

NB - I commented the ATI Analytics component because it needs refactored to accept only what it needs and not the whole Arfes response.